### PR TITLE
Add CloudWatch as grafana datasource type (#44298)

### LINKED ISSUE #44298: grafana_datasource option for using cloudwatch exporter
##### SUMMARY
We would like to use ansible grafana datasource module to configure cloudwatch datasource which is not possible because of possible choices in ds_type parameter
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME

grafana_datasource
##### ANSIBLE VERSION

```
ansible 2.5.4
  config file = None
  configured module search path = [u'/Users/michalploski/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### STEPS TO REPRODUCE

```yaml

```

##### EXPECTED RESULTS

Option to use cloudwatch datasource

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by martinwangjian**
##### SUMMARY

Add CloudWatch as grafana datasource type.
Fixes #44298

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
grafana_datasource

##### ANSIBLE VERSION
Here is my ansible --version
```
ansible 2.6.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.6 (default, Jul 19 2018, 16:29:00) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

### CODE REVIEW

File: lib/ansible/modules/monitoring/grafana_datasource.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 module: grafana_datasource
 author:
   - Thierry Sallé (@seuf)
+  - Martin Wang (@Mart1nWang)
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_datasource.py]: you should put your github account here

File: lib/ansible/modules/monitoring/grafana_datasource.py [hunk 2]
```diff
@@ -183,6 +184,56 @@
...
+      ca-central-1,
+      cn-north-1, cn-northwest-1,
+      eu-central-1, eu-west-1, eu-west-2, eu-west-3,
+      sa-east-1,
+      us-east-1, us-east-2, us-gov-west-1, us-west-1, us-west-2
+    ]
+    version_added: 2.8
+  aws_credentials_profile:
+    description:
+      - Profile for AWS credentials for CloudWatch datasource type when aws_auth_type is 'credentials"
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_datasource.py]: when C(aws_auth_type) es C(credentials)

File: lib/ansible/modules/monitoring/grafana_datasource.py [hunk 3]
```diff
@@ -183,6 +184,56 @@
...
+    default: ''
+    required: false
+    version_added: 2.8
+  aws_access_key:
+    description:
+      - AWS access key for CloudWatch datasource type when aws_auth_type is 'keys"
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_datasource.py]: when C(aws_auth_type) is C(keys)

File: lib/ansible/modules/monitoring/grafana_datasource.py [hunk 4]
```diff
@@ -183,6 +184,56 @@
...
+    default: ''
+    required: false
+    version_added: 2.8
+  aws_secret_key:
+    description:
+      - AWS secret key for CloudWatch datasource type when aws_auth_type is 'keys"
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_datasource.py]: when C(aws_auth_type) is C(keys)

File: lib/ansible/modules/monitoring/grafana_datasource.py [hunk 5]
```diff
@@ -183,6 +184,56 @@
...
+    default: ''
+    required: false
+    version_added: 2.8
+  aws_assume_role_arn:
+    description:
+      - AWS IAM role arn to assume for CloudWatch datasource type when aws_auth_type is 'arn"
```
Comment (seuf) [inline lib/ansible/modules/monitoring/grafana_datasource.py]: when C(aws_auth_type) is C(arn)

**Review by seuf — CHANGES_REQUESTED** [review]
Some documentation issues. The rest looks good

**Comment by martinwangjian** [comment]
Thx! all documentation issues are fixed

**Review by resmo — APPROVED** [review]
shipit

**Comment by resmo** [comment]
closes #44298

**Comment by resmo** [comment]
Thanks!

**Pull request merged**